reporting/bootstrap.js

```diff
@@ -0,0 +1,115 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global APP_SHUTDOWN:false */
+
+"use strict";
+
+const {classes: Cc, interfaces: Ci, utils: Cu} = Components;
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/Timer.jsm");
+Cu.import("resource://gre/modules/Log.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "TelemetryController",
+  "resource://gre/modules/TelemetryController.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "TelemetryEnvironment",
+  "resource://gre/modules/TelemetryEnvironment.jsm");
+
+const kTelemetryPingDelay = 15000;
+
+// Preferences this add-on uses.
+const kPrefPrefix = "extensions.searchvolmodel.";
+const PREF_GUID = `${kPrefPrefix}guid`;
+const PREF_LOGGING = `${kPrefPrefix}logging`;
+const PREF_TELEMETRY_SENT = `${kPrefPrefix}additionaltelemetrysent`;
+const PREF_DISTRIBUTION_ID = "distribution.id";
+
+
+const kExtensionID = "searchvolmodel@mozilla.com";
+
+// Logging
+const log = Log.repository.getLogger("extensions.searchvolmodelextra.bootstrap");
+log.addAppender(new Log.ConsoleAppender(new Log.BasicFormatter()));
+// Useful log levels: All = -1, Warn = 50, Info = 40, Debug = 20, Trace = 10, see Log.jsm
+log.level = Services.prefs.getIntPref(PREF_LOGGING, Log.Level.Warn);
+
+/**
+ * Called when the add-on is installed.
+ *
+ * @param {Object} data Data about the add-on.
+ * @param {Number} reason Indicates why the extension is being installed.
+ */
+function install(data, reason) {
+  // Nothing specifically to do, startup will set everything up for us.
+}
+
+/**
+ * Called when the add-on is uninstalled.
+ *
+ * @param {Object} data Data about the add-on.
+ * @param {Number} reason Indicates why the extension is being uninstalled.
+ */
+function uninstall(data, reason) {
+}
```

Comment:
Shouldn't the addon run Services.prefs.clearUserPref(PREF_TELEMETRY_SENT) here to clean up after itself?